src/pages/lesson/upload.tsx

```diff
@@ -48,29 +49,29 @@ const upload: FunctionComponent = () => {
     }
 
     const { data } = await axios
-      .post<ResourceApiResponse>("/api/lesson", formData)
+      .post<ApiResponse<{ lesson: ILesson }>>("/api/lesson", formData)
       .catch((error: AxiosError) => {
-        const response: ResourceApiResponse = {
+        const response: ApiResponse<{ lesson: ILesson }> = {
           success: false,
           error: getAxiosErrorMessage(error),
         };
         return { data: response };
       });
 
-    const { success } = data;
+    const success = data.success;
 
     if (success) {
-      const { id } = data;
+      const _id = data?.data?.lesson._id;
       const message = (
         <span>
           Leçon {isDraft ? "sauvegardée" : "créée"} avec succès !
-          <Alert.Link href={`/api/lesson/${id}`}>(Voir la leçon)</Alert.Link>
+          <Alert.Link href={`/lesson/${_id}`}>(Voir la leçon)</Alert.Link>
         </span>
       );
 
       dispatch(addAlert({ message, success }));
     } else {
-      const { error } = data;
+      const error = data.error;
       dispatch(
         addAlert({
           message: `Création de la leçon échouée : ${error}`,
```

Comment:
```suggestion
          message: `Création de la leçon échouée : ${data.error}`,
```
Et tu évites de déclarer une variable en plus comme ça ^^

Reply:
Done